### ISSUE TITLE
Invalid constraint error since 1.2

### ISSUE THREAD

**Issue by mkonterS** [python-poetry/poetry#7065]
- **Poetry version**: Poetry (version `1.2.2`)
- **Python version**: Python `3.10.8`
- **OS version and name**: `Linux delta 5.15.79-1-lts #1 SMP Wed, 16 Nov 2022 11:03:21 +0000 x86_64 GNU/Linux` (Arch Linux)
- **pyproject.toml**:
```
  [tool.poetry]
name = "showcase"
version = "0.1.0"
description = ""
authors = ["Your Name <you@example.com>"]
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.10"
cloudevents = "^1.7.0"


[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue
With the `pyproject.toml` file provided above I get the following error (With `1.1.15` I did not had this error):

```
$ poetry lock
Updating dependencies
Resolving dependencies... (0.1s)Invalid constraint (pydantic (>=1.0.0<1.9.0) ; (python_version <= "3.6") and extra == 'pydantic') found in cloudevents-1.7.0 dependencies, skipping
Invalid constraint (pydantic (>=1.0.0<2.0) ; (python_version > "3.6") and extra == 'pydantic') found in cloudevents-1.7.0 dependencies, skipping
Resolving dependencies... (0.2s)

Writing lock file
```
With debug:
```
$ poetry lock -vvv
Loading configuration file /home/max/.config/pypoetry/config.toml
Loading configuration file /home/max/.config/pypoetry/auth.toml
Using virtualenv: /home/max/Documents/playground/poetry_broken/.venv
Project environment contains an empty path in sys_path, ignoring.
Updating dependencies
Resolving dependencies...
   1: fact: showcase is 0.1.0
   1: derived: showcase
   1: fact: showcase depends on cloudevents (^1.7.0)
   1: selecting showcase (0.1.0)
   1: derived: cloudevents (>=1.7.0,<2.0.0)
[keyring.backend] Loading KWallet
[keyring.backend] Loading SecretService
[keyring.backend] Loading Windows
[keyring.backend] Loading chainer
[keyring.backend] Loading libsecret
[keyring.backend] Loading macOS
Creating new session for pypi.org
Source (PyPI): 1 packages found for cloudevents >=1.7.0,<2.0.0
Invalid constraint (pydantic (>=1.0.0<1.9.0) ; (python_version <= "3.6") and extra == 'pydantic') found in cloudevents-1.7.0 dependencies, skipping
Invalid constraint (pydantic (>=1.0.0<2.0) ; (python_version > "3.6") and extra == 'pydantic') found in cloudevents-1.7.0 dependencies, skipping
   1: fact: cloudevents (1.7.0) depends on deprecation (>=2.0,<3.0)
   1: selecting cloudevents (1.7.0)
   1: derived: deprecation (>=2.0,<3.0)
Source (PyPI): 9 packages found for deprecation >=2.0,<3.0
   1: fact: deprecation (2.1.0) depends on packaging (*)
   1: selecting deprecation (2.1.0)
   1: derived: packaging
[urllib3.connectionpool] Starting new HTTPS connection (1): pypi.org:443
[urllib3.connectionpool] https://pypi.org:443 "GET /simple/packaging/ HTTP/1.1" 304 0
Source (PyPI): 38 packages found for packaging *
   1: fact: packaging (21.3) depends on pyparsing (>=2.0.2,<3.0.5 || >3.0.5)
   1: selecting packaging (21.3)
   1: derived: pyparsing (>=2.0.2,!=3.0.5)
[urllib3.connectionpool] https://pypi.org:443 "GET /simple/pyparsing/ HTTP/1.1" 304 0
Source (PyPI): 39 packages found for pyparsing >=2.0.2,<3.0.5 || >3.0.5
   1: selecting pyparsing (3.0.9)
   1: Version solving took 0.182 seconds.
   1: Tried 1 solutions.
```

**Comment by dimbleby** [comment]
what part of this makes you think there's a poetry bug?

poetry has encountered an invalid constraint, told you what package it found that on, and told you what the invalid constraint it, and carried on and worked anyway.  What more do you want?!

**Comment by mkonterS** [comment]
@dimbleby as this worked with Poetry `1.1.15` and I only upgraded poetry since I last run `poetry lock` I was under the assumption that this is a bug on the side of Poetry.
If this isn't like that, that I apologize for that

**Comment by neersighted** [comment]
Not a bug; Poetry implicitly converting invalid constraints to the wildcard constriant was fixed in 1.2.x. A bug should be filed against the buggy package.
